executor/builder.go

```diff
@@ -1168,12 +1179,15 @@ func (b *executorBuilder) getStartTS() (uint64, error) {
 	}
 
 	startTS := b.ctx.GetSessionVars().SnapshotTS
-	if startTS == 0 && b.ctx.Txn(true).Valid() {
-		startTS = b.ctx.Txn(true).StartTS()
+	txn, err := b.ctx.Txn(true)
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+	if startTS == 0 && txn.Valid() {
```

Comment:
we need `b.ctx.Txn(true)` even if `b.ctx.GetSessionVars().SnapshotTS > 0` ?

Reply:
You're right, we don't need it.
However, it's not a bug to active the transaction here, as long as the startTS is get from `SnapshotTS`